core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -1968,4 +1970,87 @@ public void dropNonEmptyNamespace() {
         .as("Namespace should not exist")
         .isFalse();
   }
+
+  @Test
+  public void registerView() {
+    C catalog = catalog();
+
+    // Register view is not yet supported for REST catalog
+    assumeThat(catalog instanceof RESTCatalog).isFalse();
+
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(identifier.namespace());
+    }
+
+    View originalView =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withDefaultCatalog(catalog().name())
+            .withQuery("spark", "select * from ns.tbl")
+            .withProperty(GC_ENABLED, "false")
```

Comment:
why do we need this flag? It doesn't seem to be required when testing table registration

Reply:
The drop view will delete the view metadata. So, we will not have any file to register. Hence, needed to just delete entry from catalog and use file to register again.